modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/credential/CredentialCenter.java

```diff
@@ -201,11 +250,89 @@ public CredentialDefinition getCredential(Csp csp, String userName) {
+                            inputCredential.getType(), csp));
+        }
+        Set<String> filledVariableNameSet =
+                inputCredential.getVariables().stream().map(CredentialVariable::getName)
+                        .collect(Collectors.toSet());
+        // check all fields in the input credential are valid based on the defined credentials.
+        for (AbstractCredentialInfo credentialDefinition : credentialDefinitions) {
+            CredentialDefinition credential = (CredentialDefinition) credentialDefinition;
+            Set<String> needVariableNameSet =
+                    credential.getVariables().stream().map(CredentialVariable::getName)
+                            .collect(Collectors.toSet());
+            if (StringUtils.equals(credential.getName(), inputCredential.getName())
+                    && filledVariableNameSet.containsAll(needVariableNameSet)) {
+                Set<String> blankValueFiledNames = new HashSet<>();
+                for (CredentialVariable credentialVariable : inputCredential.getVariables()) {
+                    if (StringUtils.isBlank(credentialVariable.getValue())) {
+                        blankValueFiledNames.add(credentialVariable.getName());
+                    }
+                }
+                if (blankValueFiledNames.size() > 0) {
+                    throw new IllegalArgumentException(
+                            String.format(
+                                    "Defined credentials required fields %s value is blank.",
+                                    blankValueFiledNames));
+
+                }
+                return;
+            }
+        }
+        throw new IllegalArgumentException(
+                String.format("Defined credentials provided by Csp:%s with name:%s not found.",
+                        csp, inputCredential.getName()));
+
+    }
+
+    /**
+     * Create credential for the @Csp with @userName.
      *
-     * @param id The id of the deployed service.
-     * @return Returns DB entity of the deployed service.
+     * @param csp        The cloud service provider.
+     * @param userName   The userName to create for.
+     * @param credential The credential to create.
+     * @return true of false.
      */
+    public boolean createCredential(Csp csp, String userName, CredentialDefinition credential) {
+        String key = getCacheKey(csp, userName);
+        credentialCacheManager.putCache(key, credential);
+        clearExpiredCache();
+        return true;
+    }
+
+    private CredentialDefinition fillCredentialWithDbEntity(CreateCredential createCredential,
+                                                            DeployServiceEntity deployService) {
+        createCredential.setCsp(deployService.getCsp());
+        createCredential.setUserName(deployService.getUserName());
+        return transCredential(createCredential);
+
+    }
+
+    private CredentialDefinition transCredential(CreateCredential createCredential) {
```

Comment:
createCredentialDefinitionObject is a better name here

Reply:
Ok.